vcd/resource_vcd_vapp_vm_boot_options_test.go

```diff
@@ -0,0 +1,562 @@
+//go:build vapp || vm || ALL || functional
+
+package vcd
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/vmware/go-vcloud-director/v2/govcd"
+)
+
+func TestAccVcdVAppVmBootOptions(t *testing.T) {
+	preTestChecks(t)
+	var (
+		vapp     govcd.VApp
+		vm       govcd.VM
+		vappName string = t.Name()
+		vmName   string = t.Name() + "-vm"
+	)
+
+	if testConfig.VCD.ProviderVdc.StorageProfile == "" || testConfig.VCD.ProviderVdc.StorageProfile2 == "" {
+		t.Skip("Both variables testConfig.VCD.ProviderVdc.StorageProfile and testConfig.VCD.ProviderVdc.StorageProfile2 must be set")
+	}
+
+	var params = StringMap{
+		"Org":                    testConfig.VCD.Org,
+		"Vdc":                    testConfig.Nsxt.Vdc,
+		"EdgeGateway":            testConfig.Networking.EdgeGateway,
+		"CatalogName":            testConfig.VCD.Catalog.NsxtBackedCatalogName,
+		"VappTemplateName":       testConfig.VCD.Catalog.CatalogItemWithEfiSupport,
+		"VAppName":               vappName,
+		"VAppNameWithTemplate":   vappName + "-template",
+		"VappVMName":             vappName,
+		"VappVMWithTemplateName": vappName + "-template",
+		"EmptyVMName":            vmName,
+		"VMWithTemplateName":     vmName + "-template",
+		"Tags":                   "vapp vm",
+	}
+	testParamsNotEmpty(t, params)
+
+	params["FuncName"] = t.Name()
+	configText1 := templateFill(testAccCheckVcdVAppVmBootOptions, params)
+	debugPrintf("#[DEBUG] CONFIGURATION: %s\n", configText1)
+
+	params["FuncName"] = t.Name() + "-step1"
+	configText2 := templateFill(testAccCheckVcdVAppVmBootOptionsStep1, params)
+	debugPrintf("#[DEBUG] CONFIGURATION: %s\n", configText2)
+
+	params["FuncName"] = t.Name() + "-step2"
+	configText3 := templateFill(testAccCheckVcdVAppVmBootOptionsStep2, params)
+	debugPrintf("#[DEBUG] CONFIGURATION: %s\n", configText3)
+
+	if vcdShortTest {
+		t.Skip(acceptanceTestsSkipped)
+		return
+	}
+
+	vcd := createTemporaryVCDConnection(false)
+	if vcd.Client.APIVCDMaxVersionIs("<37.1") {
+		t.Skip("Most boot options are only available since 37.1")
+	}
```

Comment:
Should this check be set at the top, before any `templateFill` calls that generate binary tests? Binary tests would fail in 10.4.0

Reply:
True, updated.